### DISCUSSION TITLE
Additional esc key (Caps Lock) in Linux Mint does not work with recording macros

### DISCUSSION THREAD

**Discussion by kristian-clausal** [helix-editor/helix#8335] [Troubleshooting]
### Summary

I'm using Caps Lock as an alternative ESC key on Linux Mint, set in Keyboard Preferences -> Layouts -> Options -> Caps Lock behavior.

I am doing a repetitive task and decided to record a macro, and it took me a while to figure out that what was going wrong with the process (no macro was being recorded, pressing Q again started a new recording) was that I was using Caps Lock-as-ESC to exit Insert mode. Exiting with the physical ESC key worked as expected.

### Platform

Linux Mint

### Terminal Emulator

Mate terminal

### Helix Version

23.05-181-gd3f8e059

**Comment by the-mikedavis** [comment]
Is there a difference in the keycodes between actual escape and the remapped escape if you run something like `showkey -a`? Ideally also try cloning [crossterm](https://github.com/crossterm-rs/crossterm) (the terminal manipulation library we use) and run the `event-read` example: `cargo run --example event-read` and see if both versions of escape can quit the example.

AFAIK we don't do any special key handling when in a macro vs. not in a macro so I would expect that either version of escape should work the same whether you're recording a macro or not.
